File: 백준/Silver/1072. 게임/게임.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
X, Y = map(int, input().split())
Z = Y * 100 // X #승률
if Z >= 99: ##99가 넘어가버리면 절대 못이김
    print(-1)
else:
    answer = 0
    left = 1
    right = X
 
    while left <= right:
        mid = (left + right)//2 
        if (Y+mid)*100 // (X+mid) <= Z:
            left = mid+1
        else:
            answer = mid
            right = mid - 1
 
    print(answer)

File: 백준/Silver/1072. 게임/게임.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var x, y int
	fmt.Fscanf(reader, "%d %d\n", &x, &y)
	z := int(y * 100 / x)

	if z >= 99 {
		fmt.Fprintf(writer, "%d", -1)
	} else {
		result := -1
		left := 1
		right := x
		for left <= right {
			mid := (left + right) / 2
			if int((y+mid)*100/(x+mid)) <= z {
				left = mid + 1
			} else {
				result = mid
				right = mid - 1
			}
		}
		fmt.Println(result)
	}
}
